### PR TITLE
Adding regex as a pure avoid for LC

### PR DESCRIPTION
**Pull request by Denubis**
Just had to manually check through the data intro lesson, so I've added it as a "pure avoid."

### CODE REVIEW

**Comment by zpainter** [comment]
Perhaps the spreadsheets (tidy data) wouldn't be truly terrible compared to anything else on the docket, although I would agree that this lesson family in general should be avoided for demos. It is very domain and audience specific.

File: _extras/demo_lessons.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ If you are an instructor in training and wish to use lessons from Data Carpentry
...
 | [Library Carpentry: OpenRefine](https://librarycarpentry.org/lc-open-refine/) | [Importing data into OpenRefine](https://librarycarpentry.org/lc-open-refine/02-importing-data/index.html) | any other episode | [Introduction to OpenRefine](https://librarycarpentry.org/lc-open-refine/01-introduction/index.html) has no live coding. <br/> Later episodes have dependencies. |
-
+| [Library Carpentry: Library Carpentry: Introduction to Working with Data (Regular Expressions)](https://librarycarpentry.org/lc-data-intro/) | No episodes | Avoid all episodes. | Episodes are entirely exercises and a poor fit for live coding demos |
```
Comment (brownsarahm) [inline _extras/demo_lessons.md:68]:
```suggestion
| Library Carpentry: Introduction to Working with Data (Regular Expressions)](https://librarycarpentry.org/lc-data-intro/) | No episodes | Avoid all episodes. | Episodes are reference with exercises and a poor fit for live coding demos |
```

**Comment by brownsarahm** [comment]
@zpainter I think adding the tidy data as an allowable lesson is likely a good idea, but separate from avoiding this lesson.  That is related to #874 which has slowed discussion, but maybe is worth reviving over there and letting this get merged in, since it is a core lesson that we want trainees to avoid.

File: _extras/demo_lessons.md (hunk 1 above)
Comment (Denubis) [reply _extras/demo_lessons.md]: I tried another formulation of the avoid. Fixed the double header. I'm going to resolve this convo. Take a look at my edit to see if it works?

**Comment by brownsarahm** [comment]
LGTM, thanks! @Denubis

**Pull request merged**